baremetal/metal3machine_manager.go

```diff
@@ -958,6 +1134,17 @@ func (m *MachineManager) setHostConsumerRef(ctx context.Context, host *bmh.BareM
 	}
 	host.OwnerReferences = hostOwnerReferences
 
+	// Delete nodeReuseLabelName from host
+	m.Log.Info("Deleting nodeReuseLabelName from host, if any")
```

Comment:
why are we deleting the label in `setHostConsumerRef` ?

Reply:
This was chosen as **'best'** time to remove the label while using node reuse feature because once we set `consumerRef` which means M3M will be linked to BMH with nodeReuse label, and so we do not need the label anymore. Do you foresee any problems with that @kashifest ?